internal/dependency/manager/manager_test.go

```diff
@@ -0,0 +1,288 @@
+package manager
+
+import (
+	"github.com/g2a-com/klio/internal/context"
+	"github.com/g2a-com/klio/internal/dependency"
+	"github.com/spf13/afero"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+//==========================================================================================
+//::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+//==========================================================================================
+// Take a look at DEP INDEX in utils_test.go
+//==========================================================================================
+
+func TestGetInstalledCommands(t *testing.T) {
+	tests := []struct {
+		// Name of the test
+		name string
+		// Injected mock function for returning local dependencies
+		fetchIndexFunction func(filePath string) (*dependency.DependenciesIndex, error)
+		// Configuration paths for klio
+		paths context.Paths
+		// Desired index list of dependencies installed
+		want []dependency.DependenciesIndexEntry
+	}{
+		{
+			name:               "errorFromIndex",
+			fetchIndexFunction: fetchInvalidIndex,
+			paths: context.Paths{
+				ProjectInstallDir: validProjectInstallPath,
+				GlobalInstallDir:  validGlobalInstallPath,
+			},
+			want: nil,
+		},
+		{
+			name:               "emptyDirConfig",
+			fetchIndexFunction: fetchInvalidIndex,
+			paths: context.Paths{
+				ProjectInstallDir: "",
+				GlobalInstallDir:  "",
+			},
+			want: nil,
+		},
+		{
+			name:               "simpleValidIndex",
+			fetchIndexFunction: fetchSimpleValidIndex,
+			paths: context.Paths{
+				ProjectInstallDir: validProjectInstallPath,
+				GlobalInstallDir:  validGlobalInstallPath,
+			},
+			want: properlyProcessedValidSimpleIndex,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mgr := &Manager{
+				fetchDependencyIndex: tt.fetchIndexFunction,
+			}
+			if got := mgr.GetInstalledCommands(tt.paths); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetInstalledCommands() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+//==========================================================================================
+//::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+//==========================================================================================
+// Take a look at REGISTRY MOCKS in utils_test.go
+//==========================================================================================
+
+func TestGetUpdateFor(t *testing.T) {
+	tests := []struct {
+		// Name of the test
+		name string
+		// Dependency to get update for
+		dep dependency.Dependency
+		// Desired list of versions available
+		want Updates
+		// Flag marking that the test should end up with error
+		wantErr bool
+	}{
+		{
+			name: "nonExistentRegistry",
+			dep: dependency.Dependency{
+				Registry: "nonExistent",
+			},
+			want:    Updates{},
+			wantErr: true,
```

Comment:
It may be out of scope of this PR, but checking for *any* error is not enough, we need to check if returned error is the exact one we expect.